### PR TITLE
crypto: Rename PreviouslyVerified to VerificationViolation

### PR DESCRIPTION
**Pull request by andybalaam**
Part of https://github.com/element-hq/element-meta/issues/2526

For consistency with other places, we have now settled on `VerificationViolation` as the best way to express this situation.

- [x] Public API changes documented in changelogs (optional)

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs [hunk 1]
```diff
@@ -273,8 +269,9 @@ pub enum SenderDataType {
     UnknownDevice = 1,
     /// The [`SenderData`] is of type `DeviceInfo`.
     DeviceInfo = 2,
-    /// The [`SenderData`] is of type `SenderUnverifiedButPreviouslyVerified`.
-    SenderUnverifiedButPreviouslyVerified = 3,
+    /// The [`SenderData`] is of type
+    /// `VerificationViolation`.
```
Comment (bnjbvr) [inline crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs:273]: nit: doesn't this fit on a single line?

**Review by bnjbvr — APPROVED** [review]
LGTM. Note that `previously_verified` returns quite a few matches with a plain grep, so consider whether you want to rename those also or not (and no need to ask for another review from me IMO, unless you're unsure/unclear about one thing in particular).

File: crates/matrix-sdk-crypto/CHANGELOG.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ Breaking changes:
   previously-verified, unverified and verified.
   ([#3877](https://github.com/matrix-org/matrix-rust-sdk/pull/3877))
 
-- Where `EncryptionInfo` is returned it may include the new `PreviouslyVerified`
+- Where `EncryptionInfo` is returned it may include the new `VerificationViolation`
```
Comment (richvdh) [inline crates/matrix-sdk-crypto/CHANGELOG.md:72]: driveby comment: given we've already released this stuff, can you put a separate changelog entry higher up, please?

File: crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs (hunk 1 above)
Comment (andybalaam) [reply crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs]: Fixed in b1e946c0fc

File: crates/matrix-sdk-crypto/CHANGELOG.md (hunk 2 above)
Comment (andybalaam) [reply crates/matrix-sdk-crypto/CHANGELOG.md:72]: Thanks, I dithered about this and chose the wrong choice!

Comment (andybalaam) [reply crates/matrix-sdk-crypto/CHANGELOG.md]: Fixed in a3da650b06
